### PR TITLE
Fix/commit datatype

### PR DESCRIPTION
**Pull request by bplatz**
Most non-standard datatypes were not getting properly written into commit files, causing values for datatypes like `@json` to load as xsd:string.

Also addresses:
- `@json` was being converted to xsd:json which isn't an actual xsd type. now retains `@json` type
- unsigned integer types were being coerced into their base types (e.g. xsd:unsignedShort was being coerced to java Short). Because java doesn't have unsigned native types, it needs to upsize the Java type... in this example it should coerce to a java Integer.
- untyped decimals were being coerced into xsd:decimal, but should have been coerced to xsd:double. Turtle can coerce untyped decimals into xsd:decimal -or- xsd:double (if using `e` notation), but JSON-LD can only coerce into xsd:double
- in commit files, we did some acrobatics to track 'refs' and write `@type: @id` into context for refs, but it wasn't being done thoroughly and could be inaccurate. Also we are now used to seing `{"@id" ...}` for nodes in JSON-LD. This is removed and will always be accurate now
- in commit files, we wrote out `{"@value" ...}` even when no `@type` was written. This is not needed, takes extra disk space, and isn't the common way of seeing inferred values in JSON-LD, so now if using an inferred data type we just write the value, not a `@value` map.

This closes https://github.com/fluree/db/issues/884

### CODE REVIEW

**Review by zonotope — APPROVED** [review]
🪶

File: src/clj/fluree/db/constants.cljc [hunk 1]
```diff
@@ -71,6 +71,7 @@
 (def ^:const iri-value "@value")
 (def ^:const iri-language "@language")
 (def ^:const iri-type "@type")
+(def ^:const iri-json "@json")
```
[the remark this answers is not in the document]
Comment (bplatz) [reply src/clj/fluree/db/constants.cljc]: I hadn't realized. I've updated it so it does this.

**Comment by bplatz** [comment]
Also, I found that if you used a custom data type (not in our defaults list), deserialization from disk wouldn't work properly. This is addressed in e38203cfee9af402fd6cd55ec85ad0267e7782c3

**Comment by bplatz** [comment]
Also added quick fix for custom datatypes that are not scalar values to throw an exception. The only non-scalar values allowed for `@value` values are for `@type: @json`.

**Pull request merged**